test/test_autograd.py

```diff
@@ -7765,6 +7764,26 @@ def test_out_variant(self):
             with self.assertRaisesRegex(RuntimeError, "out= function"):
                 torch.add(foo, bar, out=bar)
 
+    def test_mkldnn_convolution(self):
+        n = 2
+        c = 3
+        o = 4
+        x = 5
+
+        inp = torch.rand(n, c, x, dtype=torch.float, requires_grad=True)
```

Comment:
I'm not sure your inputs are correct here. For a 2d conv, you want a 4d input, and 4d weights.
Also the last two dimensions of the input should not match the weights otherwise otherwise padding/stride will be ignored.

I feel like an input of shape (2, 2, 5, 5), weight (3, 2, 3, 3), padding (1, 1), dilation (1, 1) and groups 1 would be better

Reply:
Just want to check - for weight, should it be (2, 2, 3, 3) since the first dimension should match input?